### PR TITLE
"add shortcut" button

### PR DESCRIPTION
**Pull request by Jayian1890**
A simple method of adding a new shortcut to an external executable that's accessible via the programs list.

### CODE REVIEW

**Review by blackxfiied — APPROVED** [review]
Works well, and it is spamproofed (only one entry for a single path at a time)

**Comment by BND10706** [comment]
I tested this as well. Works great. Helped me to get an application running.

File: Whisky/Utils/Wine.swift [hunk 1]
```diff
@@ -252,6 +252,11 @@ class Wine {
         return try runWineserver(["-k"], bottle: bottle)
     }
 
+    static func addShortcut(name: String, url: URL, bottle: Bottle) async throws {
```
Comment (IsaacMarovitz) [inline Whisky/Utils/Wine.swift:255]: This function does not make any use of Wien and shouldn't be implemented in this class

File: Whisky/Views/Bottle Views/BottleView.swift [hunk 2]
```diff
@@ -131,6 +131,34 @@ struct BottleView: View {
...
+                                                 UTType(importedAs: "com.microsoft.msi-installer")]
+                    panel.directoryURL = bottle.url.appendingPathComponent("drive_c")
+                    panel.begin { result in
+                        programLoading = true
+                        Task(priority: .userInitiated) {
+                            if result == .OK {
+                                if let url = panel.urls.first {
+                                    do {
+                                        let name = url.lastPathComponent
+                                        try await Wine.addShortcut(name: name, url: url, bottle: bottle)
```
Comment (IsaacMarovitz) [inline Whisky/Views/Bottle Views/BottleView.swift:149]: Validation for bottle shortcuts should be added. Removing or renaming the source file will currently not remove the dead shortcut, and will just log errors.

**Review by IsaacMarovitz — CHANGES_REQUESTED** [review]

File: Whisky/Utils/Wine.swift (hunk 1 above)
Comment (blackxfiied) [reply Whisky/Utils/Wine.swift:255]: true, should have noted that that looks out of place

Comment (Jayian1890) [reply Whisky/Utils/Wine.swift:255]: All those classes were static. That's why I figured it fit lol. I'll push a change tonight or tomorrow.

File: Whisky/Views/Bottle Views/BottleView.swift (hunk 2 above)
Comment (Jayian1890) [reply Whisky/Views/Bottle Views/BottleView.swift:149]: I could add a check to updateStartMenu(), so shortcuts are verified upon updates. Would that be sufficient? I don't want to stray too far from your preferred coding guidelines.

Comment (Jayian1890) [reply Whisky/Views/Bottle Views/BottleView.swift:149]: > Validation for bottle shortcuts should be added. Removing or renaming the source file will currently not remove the dead shortcut, and will just log errors.

I could add a check to updateStartMenu(), so shortcuts are verified upon updates. Would that be sufficient? I don't want to stray too far from your preferred coding guidelines.

**Pull request closed without merging**